src/main/java/org/opensearch/ad/model/AnomalyResult.java

```diff
@@ -262,6 +393,44 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         if (modelId != null) {
             xContentBuilder.field(MODEL_ID_FIELD, modelId);
         }
+
+        // output extra fields such as attribution and expected only when this is an anomaly
+        if (anomalyGrade != null && anomalyGrade > 0) {
+            if (totalUpdates != null) {
+                xContentBuilder.field(TOTAL_UPDATES_FIELD, totalUpdates);
+            }
+            if (startOfAnomaly != null) {
+                xContentBuilder.field(START_OF_ANOMALY_FIELD, startOfAnomaly);
+            }
+            if (inHighScoreRegion != null) {
+                xContentBuilder.field(IN_HIGH_SCORE_REGION_FIELD, inHighScoreRegion);
+            }
+            if (relativeIndex != null) {
+                xContentBuilder.field(RELATIVE_INDEX_FIELD, relativeIndex);
+            }
+            if (currentTimeAttribution != null) {
+                xContentBuilder.array(CURRENT_TIME_ATTRIBUTION_FIELD, currentTimeAttribution);
+            }
+            if (oldValues != null) {
+                xContentBuilder.array(OLD_VALUES_FIELD, oldValues);
+            }
+            // In OpenSearch, an array of arrays like [ 1, [ 2, 3 ]] is the equivalent of [ 1, 2, 3 ]
+            // Also, even though expectedValuesList if a two-dimensional array, it has only one nested
+            // array inside in current implementation.
```

Comment:
Makes sense to have 2D array for future extensibility. Just thinking, in terms of index readability, it could get confusing if later on there is more fields added per expected value (like different confidences, etc.). When indexing, each new field per expected value would add `n` new values to the end of the array, where `n` is the expected value / feature length.

Since there's current efforts to expose the result index to the user, this could make it extra confusing for the user to understand the indexing strategy done here. Did you consider using nested fields, and just index an array of expected value objects? Currently, it would only contain a single `value` field, but if that was to be expanded on in the future, could just add more fields?

Reply:
Can you give me an example how nested fields work? Say I have an expected value array [1, 2, 3, 4, 5].  Tomorrow there are two expected value array with different confidence. I need to add [7, 8, 9, 10, 11].  How do I change mapping?

Reply:
I was thinking of it in such a way, that each object in the array would contain all of the expected value info for each corresponding feature.

For example, given features `f1` and `f2`, could have expected values like the following:
```
[
  // info about feature f1
  {
    expectedValue1: 5,
    confidence1: 0.95,
    expectedValue2: 7,
    confidence2: 0.82
  },

  // info about feature f2
  {
    expectedValue1: 20,
    confidence1: 0.90,
    expectedValue2: 16,
    confidence2: 0.80
  }  
]
```

Not sure if this is the right way to go either, though, since changing the mappings in the future may lead to other issues. Your solution is obviously simpler, just thinking about user readability in the future when we open up these results publicly.

Reply:
Changing the mapping in the future would not be easy. Also, searching with nested fields can be difficult (e.g., recall our discussions about entity field and how complicated of the query behind our facet API is).  So I didn't start with nested fields.

If you think querying nested fields is not a problem, I am willing to go that route.  I am thinking of two ways to work around the changeability issue.  One is to reserve expectedValue1~expectedValue20 in the mapping.  As long as we don't allocate more than 20 expected values (unlikely), we are fine.

Another is to have sth like:

[
  {
    expectedValue: [1,2,3],
    likelihood: 0.95
  },
  {
    expectedValue: [11,12,13],
    likelihood: 0.90
  }  
]

What do you think?

Reply:
I'd suggest the nested fields .. in the above the idea is f1 = 1, f2= 2, and f3 = 3 in a scenario with likelihood 0.95 and f1=11, f2=12, and f3=13 in another scenario with likelihood 0.90 (let's not worry about things summing to 1, etc. for this discussion). What this captures is that in either scenarios the f1-f3 gap is in a narrow range. That correlation is lost if we perform a field by field analysis -- separating the confidence, expected value of each field from other fields. If we were interested in single fields -- we should perform 3 different anomaly detections (one for each field). One performs a (multivariate) analysis on several fields together if only there is something to be gained (in this case the gain is the correlated/conditional expectation).